File: JUnit/saucery-core/src/main/java/au/com/fullcirclesolutions/saucery/utils/WebControl.java
package au.com.fullcirclesolutions.saucery.utils;

import java.util.List;
import org.openqa.selenium.By;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.remote.RemoteWebDriver;

public class WebControl {
    private final By _by;

    public WebControl(By by) {
        _by = by;
    }

    public WebElement Find(RemoteWebDriver driver) {
        return driver.findElement(_by);
    }

    public List<WebElement> FindAll(RemoteWebDriver driver) {
        return driver.findElements(_by);
    }
}
/*
 * Copyright Andrew Gray, Full Circle Solutions
 * Date: 14th August 2014
 */

File: NUnit/Version2/Saucery2/Util/WebControl.cs
using System.Collections.ObjectModel;
using OpenQA.Selenium;
using OpenQA.Selenium.Remote;

namespace Saucery2.Util {
    public class WebControl {
        private readonly By _by;

        public WebControl(By by){
            _by = by;
        }

        public IWebElement Find(RemoteWebDriver driver){
            return driver.FindElement(_by);
        }

        public ReadOnlyCollection<IWebElement> FindAll(RemoteWebDriver driver){
            return driver.FindElements(_by);
        }
    }
}
/*
 * Copyright Andrew Gray, Full Circle Solutions
 * Date: 14th August 2014
 * 
 */